src/report-embedded-add/report-embedded-add.controller.js

```diff
@@ -0,0 +1,81 @@
+/*
+ * This program is part of the OpenLMIS logistics management information system platform software.
+ * Copyright © 2017 VillageReach
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *  
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org. 
+ */
+
+(function() {
+    'use strict';
+    /**
+     * @ngdoc controller
+     * @name report-embedded-add.controller:ReportEmbeddedAddController
+     *
+     * @description
+     * Allows user to add new reports
+     */
+    angular
+        .module('report-embedded-add')
+        .controller('ReportEmbeddedAddController', ReportEmbeddedAddController);
+
+    ReportEmbeddedAddController.$inject = ['reportEmbeddedService', 'loadingModalService'];
+
+    function ReportEmbeddedAddController(reportEmbeddedService, loadingModalService) {
+        var vm = this;
+
+        vm.goBack = goBack;
+        vm.add = add;
+        vm.validateField = validateField;
+        vm.invalidFields = new Set();
+        vm.report = undefined;
+
+        vm.categories = ['Administration', 'Orders', 'Requistion', 'Stock'];
+
+        function add() {
+            loadingModalService.open();
+
+            if (validateAddReport()) {
+                reportEmbeddedService.add(vm.report)
+                    .then(function() {
+                        goBack();
+                    });
+            }
+
```

Comment:
This empty line looks off to me. I think it shouldn't be here, are you using a formatter for the code?